terraform/projects/app-govuk-rds/main.tf

```
@@ -0,0 +1,692 @@
+  default     = "01:00-03:00"
+}
+
+variable "backup_retention_period" {
+  type        = string
+  description = "The days to retain backups for."
+  default     = "7"
+}
+
+variable "skip_final_snapshot" {
+  type        = bool
+  description = "Set to true to NOT create a final snapshot when the cluster is deleted."
+  default     = false
+}
+
+variable "terraform_create_rds_timeout" {
+  type        = string
+  description = "Set the timeout time for AWS RDS creation."
+  default     = "2h"
+}
+
+variable "terraform_update_rds_timeout" {
+  type        = string
+  description = "Set the timeout time for AWS RDS modification."
+  default     = "2h"
+}
+
+variable "terraform_delete_rds_timeout" {
+  type        = string
+  description = "Set the timeout time for AWS RDS deletion."
+  default     = "2h"
+}
+
+variable "internal_zone_name" {
+  type        = string
+  description = "The name of the Route53 zone that contains internal records"
+}
+
+variable "internal_domain_name" {
+  type        = string
+  description = "The domain name of the internal DNS records, it could be different from the zone name"
+}
+
+variable "instance_ami_filter_name" {
+  type        = string
+  description = "Name to use to find AMI images for the instance"
+  default     = "ubuntu/images/hvm-ssd/ubuntu-trusty-14.04-amd64-server-*"
+}
+
+variable "instance_key_name" {
+  type        = string
+  description = "Name of the instance key"
+  default     = "govuk-infra"
+}
+
+locals {
+  tags = {
+    Project         = var.stackname
+    aws_stackname   = var.stackname
+    aws_environment = var.aws_environment
+  }
+}
+
+terraform {
+  backend "s3" {}
+  required_version = "= 1.1.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = var.aws_region
+}
+
+
+# Resources
+# --------------------------------------------------------------
+
+data "aws_route53_zone" "internal" {
+  name         = var.internal_zone_name
+  private_zone = true
+}
+
+# --------------------------------------------------------------
+
+resource "aws_security_group" "rds" {
+  for_each = var.databases
+
+  name        = "${var.stackname}_${each.value.name}_rds_access"
+  vpc_id      = data.terraform_remote_state.infra_vpc.outputs.vpc_id
+  description = "Access to ${each.value.name} RDS"
+
+  tags = merge(local.tags, { Name = "${var.stackname}_${each.value.name}_rds_access" })
+}
+
+resource "aws_security_group" "db_admin" {
+  for_each = var.databases
+
+  name        = "${var.stackname}_${each.value.name}_db-admin_access"
+  vpc_id      = data.terraform_remote_state.infra_vpc.outputs.vpc_id
+  description = "Access to ${each.value.name} db-admin"
+
+  tags = merge(local.tags, { Name = "${var.stackname}_${each.value.name}_db-admin_access" })
+}
+
+resource "aws_security_group_rule" "rds_ingress_db-admin_postgres" {
+  for_each = var.databases
```

I wonder if we should filter this on engine = "postgres".

Something like:

```suggestion
  for_each = {for name, config in var.databases : name => config if config.engine == "postgres"}
```

... and then we don't need to give all the postgres databases access to the myql ports and vice versa. Maybe that's too fancy - it's not really an issue if they have access to one more port which isn't even open.

---

No, that's good, I actually wanted to do that but didn't know you could